res/controllers/Hercules-DJControl-Inpulse-500-script.js

```diff
@@ -0,0 +1,1566 @@
+      value = this.inValueScale(value) * 4;
+      if (engine.isScratching(deck)) {
+        engine.scratchTick(deck, value);
+      } else {
+        engine.setValue(group, 'jog', value);
+      }
+    },
+    inputTouch: function(channel, control, value, status, group) {
+      var deck = parseInt(deckData.currentDeck.charAt(8));
+      if (this.isPress(channel, control, value, status) && this.vinylMode) {
+        engine.scratchEnable(deck,
+          this.wheelResolution,
+          this.rpm,
+          this.alpha,
+          this.beta);
+      } else {
+        engine.scratchDisable(deck);
+      }
+    },
+  });
+
+  // Loop controls
+  this.loopInButton = new components.Button({
+    midi: [0x90 + midiChannel, 0x09],
+    shiftOffset: 3,
+    shiftControl: false,
+    shiftChannel: true,
+    sendShifted: true,
+    outKey: 'loop_enabled',    // TODO: Check with loop_in?
+    unshift: function () {
+      this.inKey = 'loop_in';
+    },
+    shift: function () {
+      this.inKey = 'loop_in_goto';
+    },
+  });
+
+  this.loopOutButton = new components.Button({
+    midi: [0x90 + midiChannel, 0x0A],
+    shiftOffset: 3,
+    shiftControl: false,
+    shiftChannel: true,
+    sendShifted: true,
+    outKey: 'loop_enabled',    // TODO: Check with loop_in?
+    unshift: function () {
+      this.inKey = 'loop_out';
+    },
+    shift: function () {
+      this.inKey = 'loop_out_goto';
+    },
+  });
+
+  // Loop rotary encoder functions
+  //
+  // Push the rotary encoder
+  this.loopEncoderPush = new components.Button({
+    midi: [0x90 + midiChannel, 0x2C],
+    shiftOffset: 3,
+    shiftControl: false,
+    shiftChannel: true,
+    sendShifted: true,
+    unshift: function () {
+      this.inKey = 'reloop_toggle';
+    },
+    shift: function () {
+      this.inKey = 'beatloop_4_activate';
+    },
+  });
+
+  // Loop encoder
+  this.loopEncoder = new components.Encoder({
+    midi: [0xB0 + midiChannel, 0x0E],
+    shiftOffset: 3,
+    shiftControl: false,
+    shiftChannel: true,
+    sendShifted: true,
+    input: function (channel, control, value, status, group) {
+      // FIXME: Toggle for loop halve and double??
+      var deckGroup = deckData.currentDeck;
+      if (value >= 0x40) {
+        engine.setValue(deckGroup, "loop_halve", true);
+      } else {
+        engine.setValue(deckGroup, "loop_double", true);
+      }
+    }
+  });
+
+  // We only check and attach for slicer mode, but we have all
+  // pad buttons here if we need something extra!
+  this.padSelectButtons = [];
+  for (var i = 1; i <= 8; i++) {
+    this.padSelectButtons[i] = new components.Button({
+      midi: [0x90 + midiChannel, 0x0F + (i - 1)],
+      input: function(channel, control, value, status, group) {
+        var deck = parseInt(deckData.currentDeck.charAt(8)) - 1;
+        if (control === 0x11) {
+          DJCi500.slicerActive[deck] = true;
+        } else {
+          DJCi500.slicerActive[deck] = false;
+        }
+      },

```

Comment:
Suggestion: Replace custom code with built-in [`script.deckFromGroup` helper function](https://github.com/mixxxdj/mixxx/wiki/MIDI-Scripting#helper-functions) and prefix unused parameters with `_`.
```suggestion
      input: function(_channel, control, _value, _status, _group) {
        var deck = script.deckFromGroup(deckData.currentDeck);
        if (control === 0x11) {
          DJCi500.slicerActive[deck - 1] = true;
        } else {
          DJCi500.slicerActive[deck - 1] = false;
        }
      },
```